app/models/stage.rb

```diff
@@ -28,6 +28,8 @@ class Stage < ActiveRecord::Base
   attr_writer :command
   before_save :build_new_project_command
 
+  before_soft_delete :verify_not_part_of_pipeline
```

Comment:
So, we ok to leave this here? I don't know order of execution well enough to know where (if anywhere) else to put this.